cmd/launchBackup.go

```diff
@@ -0,0 +1,231 @@
+	if err != nil {
+		return err
+	}
+
+	// etc back up
+	etcExcludeArgs := fmt.Sprintf("cat /etc/tmpfiles.d/* | sed 's/#.*//' | awk '{print $2}' | grep '^/etc/' | sed 's#^/etc/##' > %s/etc.exclude.list; echo '.updated' >> %s/etc.exclude.list; echo 'kubernetes/manifests' >> %s/etc.exclude.list", BackupPath, BackupPath, BackupPath)
+	err = ExecuteArgs(etcExcludeArgs, BackupPath, "etc-exclude-list")
+	if err != nil {
+		log.Error(err)
+	}
+
+	// usrlocal backup
+	etcArgs := fmt.Sprintf("rsync -a %s %s", etc, path[2])
+	err = ExecuteArgs(etcArgs, path[2], etc)
+	if err != nil {
+		log.Error(err)
+	}
+
+	usrlocalArgs := fmt.Sprintf("rsync -a %s %s", usrlocal, path[3])
+	err = ExecuteArgs(usrlocalArgs, path[3], usrlocal)
+	if err != nil {
+		log.Error(err)
+	}
+
+	kubeletArgs := fmt.Sprintf("rsync -a %s %s", kubelet, path[4])
+	err = ExecuteArgs(kubeletArgs, path[4], kubelet)
+	if err != nil {
+		log.Error(err)
+	}
+
+	log.Info(strings.Repeat("-", 60))
+	log.Info("backup has succesfully finished ...")
+
+	return nil
+
+}
+
+// Cleanup deletes all old sub diectories and files in the recovery partition
+// returns: 			error
+func Cleanup(path string) error {
+	/*	//change root directory to /host
+		if err := syscall.Chroot(host); err != nil {
+			log.Errorf("Couldn't do chroot to %s, err: %s", host, err)
+			return err
+		}
+	*/
+	log.Info(strings.Repeat("-", 60))
+	log.Info("Cleaning up old contents and ostree deployment started ...")
+	log.Info(strings.Repeat("-", 60))
+	// Cleanup previous backups
+	dir, _ := os.Open(path)
+	subDir, _ := dir.Readdir(0)
+
+	// Loop over the directory's files.
+	for index := range subDir {
+		fileNames := subDir[index]
+
+		// Get name of file and its full path.
+		name := fileNames.Name()
+		fullPath := path + "/" + name
+		log.Info("\nfullpath: ", fullPath)
+
+		// Remove the file.
+		err := os.RemoveAll(fullPath)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+	}
+	log.Info("Old directories deleted with contents")
+
+	// ostree undeploy of previous deployments
+	ostreeClean := "while :; do ostree admin undeploy 1 || break; done"
+	err := ExecuteArgs(ostreeClean, "backup-time", "ostree-cleaning")
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+//CreateDir creates new sub-directories where backup will be stored
+// returns:  slice of filepath ([]string) error
+func CreateDir(path string) ([]string, error) {
+	//create backup folders
+	newPath := make([]string, len(folders))
+	os.Chdir(path)
+
+	for i := 0; i < len(folders); i++ {
+		err := os.Mkdir(folders[i], 0700)
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+		newPath[i] = path + "/" + folders[i]
+		log.Info("sub directory created at: ", newPath[i])
+	}
+	return newPath, nil
+}
+
+//ExecuteArgs execute shell commands
+//returns: 			error
+func ExecuteArgs(arg string, path string, resource string) error {
+	if resource == "etcd-cluster" {
+		_, err := exec.Command(arg, path).Output()
```

Comment:
I think we need to log the stdout/stderr of these commands, in order to debug if anything goes wrong. Otherwise, all we see in the log is "ERRO[0000] exit status 1"

Something like:
out, err := exec.Command(arg, path).CombinedOutput()
log.Printf("%s", out)